third_party/amd/lib/TritonAMDGPUTransforms/BlockPingpong.cpp

```diff
@@ -81,6 +83,22 @@ void Pingponger::appendOp(Operation *op) {
   op->moveAfter(lastInsertedOp);
   lastInsertedOp = op;
 }
+void Pingponger::moveOpAndDependencies(Operation *op) {
+  assert(lastInsertedOp != nullptr);
+  assert(lastInsertedOp->isBeforeInBlock(op));
```

Comment:
I'm not against but wonder if this is necessary here. For example, we may want to deliberately reorder some independent ops.

Reply:
I agree that's possible, but I think it breaks too many of the function's dependencies. For example, right now we have to check `isBeforeInBlock` to determine the bound of dependencies that we need to move.

I'm happy to add a TODO here, but I think it would be overkill to fully generalize this. I'll also update the function name to reflect this.